src/lineargradientbrush.c

```diff
@@ -92,9 +92,11 @@ gdip_linear_gradient_new (void)
 {
 	GpLineGradient *result = (GpLineGradient *) GdipAlloc (sizeof (GpLineGradient));
 
-	if (result && gdip_linear_gradient_init (result) != Ok) {
+	if (result) {
+		if (gdip_linear_gradient_init (result) == Ok)
+			return result;
+
 		GdipFree (result);
-		return NULL;
 	}
```

Comment:
This is not right, it will return freed object if `gdip_linear_gradient_init` fails. The previous code was correct.

Reply:
fixed